### PR TITLE
[Script] deposit into StabeX

### PR DESCRIPTION
**Pull request by fleupold**
Self explanatory

### Test Plan

Run the scripts:

```
npx truffle exec scripts/stablex/setup_environment.js 
npx truffle exec scripts/stablex/deposit.js 0 0 1
```

Then check deposit was made by using `npx truffle console`:

```
> const accounts = await web3.eth.getAccounts()
> const instance = await StablecoinConverter.deployed()
> const token_address = await instance.tokenIdToAddressMap.call(0)
> (await instance.getPendingDepositAmount(accounts[0],token_address)).toString()
'1000000000000000000'
```

### CODE REVIEW

File: scripts/stablex/deposit.js [hunk 1]
```diff
@@ -29,7 +29,9 @@ module.exports = async (callback) => {
     }
 
     await instance.deposit(token_address, amount, { from: depositor })
-    console.log("Deposit successful")
+    const tradeable_at = await instance.getPendingDepositBatchNumber(depositor, token_address)
+
+    console.log(`Deposit successful. Can be traded as of batch ${tradeable_at}`)
```
Comment (bh2smith) [inline scripts/stablex/deposit.js:34]: This is cool!

**Review by bh2smith — APPROVED** [review]
Very cool! I like the log results.

File: scripts/snapp/deposit.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const SnappAuction = artifacts.require("SnappAuction")
-const ERC20Mintable = artifacts.require("ERC20Mintable.sol")
+const ERC20 = artifacts.require("ERC20.sol")
```
Comment (anxolin) [inline scripts/snapp/deposit.js:2]: nit: I think the extension `.sol` is not needed

**Review by anxolin — CHANGES_REQUESTED** [review]
Nice we start to see scripts! this is going to be useful.

For me it doesn't migrate:
it's missing the contract: `IterableAppendOnlySet`

Maybe is missing in the `DevDependencies.sol`?

Also, I would be more in favour of using more explicit parameters, cause it helps with readability in the e2e. Also is easier to use, since we don't need to remember the order.
```bash
npx truffle exec scripts/stablex/deposit.js --account 0 --token 0 --amount 1

# or.. 
# npx truffle exec scripts/stablex/deposit.js --account-id 0 --token-id 0 --amount 1
```

Additionally set a base for further improvements of the scripts and maybe evolve them into a CLI tool. One lib that is simple to use might help with this is https://www.npmjs.com/package/yargs

Thanks for the test plan, it really makes it easier to review!

**Comment by fleupold** [comment]
@anxolin What exactly are you requesting to be changed?

I think the fact that it doesn't migrate on your end might be an issue with having to `npm i` and/or having to `rm -r build` and trying again.

I like your idea about the yargs. Would you agree it would make sense to adjust all existing script in one go, or should I start here and have them be inconsistent for now?

**Comment by anxolin** [comment]
> @anxolin What exactly are you requesting to be changed?

For some reason it failed in the 4th migration in my case. I assumed it was because it wasn't in the `DevDependencies.sol` (I checked and it wasn't there), but this wasn't the case now I deleted everything and re-tried it passes migration 4. So no change is required

> I like your idea about the yargs. Would you agree it would make sense to adjust all existing script in one go, or should I start here and have them be inconsistent for now?

I saw there's many other PRs, one per script. 
So I would say that the best now is to do it separately. I understand that for the previous dFusion it would be a considerable work to change all e2e test, but I wanted to flag this now, since we haven't write the e2e test yet

**Review by anxolin — APPROVED** [review]
After removing the `build` and `node_modules` everything worked 👍

**Comment by fleupold** [comment]
> I understand that for the previous dFusion it would be a considerable work to change all e2e test

I think it's ok, we should change it for the entire codebase. Will do that once the scripts are landed

**Pull request merged**